### ISSUE TITLE
Change github stale bot so we can keep lots of easily reproducible bugs open instead of just being closed as "not planned"

### ISSUE THREAD

**Issue by CoroNaut** [CleverRaven/Cataclysm-DDA#70275]
### Is your feature request related to a problem? Please describe.

Having easily reproducible bugs be marked as stale and then closed as "not planned" is understandably frustrating. If I put in the time to figure out how to reproduce a bug, give a full report, pictures, save file, version info, etc, it should at least get some amount of review. If it just gets closed as "not planned" after a little over 30 days, it seems like kind of a waste of time to have made it in the first place. Then what exactly are we supposed to do after that?

### Solution you would like.

4 possible solutions?
1. What if stale bot notified collaborators? They could do a quick look at the bug and at the very least put the "confirmed" tag on it so stale bot stays away from it. Perhaps after 15 days that no collaborator has commented on the issue, it will notify one to either ask questions and get more information or confirm/deny the issue.
2. Have the ability for non-collaborators to change the tags on their own post. I'm not sure of how permissions could be set or if this is possible or not. Of course this isn't ideal, some people may get tag crazy or add incorrect tags. If tag changes to a post could be confirmed or denied by collaborators, that would be ideal. 
3. Add a "Bug Report 🐛 w/ Verified Reproducible Steps" template for creating new issues that stale-bot would ignore.
4. Increase the time that stale-bot marks issues as stale to 90ish days for bug issues specifically. Suggestion posts could remain stale at 30 days or make it 60.

Solution 3 would be a fine addition in all cases. People who understand what they are doing, and can post bug reports with the knowledge that the bug is fully identified can live worry-free that the post won't just get closed as "not planned". It may get slightly abused, but I believe the gains outweigh that heavily.
Solution 1 would ensure that all bugs have some amount of attention. Suggestions could be excluded.

### Describe alternatives you have considered.

What exactly should someone do if their bug report is closed as "not planned"?
1. Live with the bug?
2. Wait until someone combs through the unknown number of closed reproducible bugs to fix them?
3. Make another issue and copy/paste the entirety of the last issue? 
4. Every 30 days, reproduce the bug and post a comment saying that it is still reproducible on the most current version? (stale bot says not to 'bump' posts).
5. Wait until someone removes or changes the feature that contained the bug?

All these alternatives are obviously not ideal.

### Additional context

I understand that the project has had times where it was up to 7k+ open issues, and that stale bot easily keeps it down. Suggestions that don't gain traction could probably be closed as "not planned", but bugs too? 
If stale bot closes this issue from being stale, I think I'd just keel over right then lol.

**Comment by ehughsbaird** [comment]
If you would like to volunteer to triage bugs, I believe this list is open.
https://github.com/CleverRaven/Cataclysm-DDA/blob/04b6eb249b041653afa14cafd7b897f621100ae3/.github/comment-commands.yml#L4

**Comment by GuardianDll** [comment]
If some bug is closed as stale it usually means that no one actively tried to confrm it, either because of rarity of bug occurrence, or because lack of knowledge what the potential issue with bug can be
It is not bad they become stale and are closed, even the opposite - having a giant amount old bugs that were fixed already, but still pile up in the list doesn't feel healthy, and constant cycle of them can reveal much more important or devastating stuff 
There is no actual way to make it any better except beforementioned volunteering to confirm bugs, or having more people around that are knowledgeable enough to fix the bug or at least point to place where the potential fix should be

**Comment by CoroNaut** [comment]
I wouldn't mind volunteering if that is available. My contribution would be limited though. Would it be against the rules to also confirm my own reproducible issues? I see the list of commands available, but not rules to using them.

I do see the problem with bugs that are fixed, but their issue not being closed. I'm sure there are very limited options to deal with that, If that is an important goal of the stale bot, I would probably say the current stale bot is fine.

More important or devastating issues wouldn't really be affected if there are constantly new issues on the same bug. Bugs that cause inconvenience or popup error messages may be noticed by a large number of players but go unreported. Like trying to repair a vehicle part, but its health not actually changing (I'm looking into that currently). It causes inconvenience, and it should probably be fixed. It's small and gets ignored by lots of players though, despite it using resources or wasting time and not yielding benefit. I'm not certain that is healthy for the game, especially for stable releases. It kind of feels like the saying "death by a thousand cuts". For example, I've found at least 50ish minor bugs in 30 days in normal game play, much more than previous years. Just how many go unreported and unresolved for all players?

Confirming a bug is a big part of keeping the issue open, which my solution 3 would remedy. Normal players that come across a bug or suggestion could make a quick issue:"I found this problem, here's my save and version". Then, more understood and reproducible bugs could just be confirmed on the spot if they put in the extra work to ensure it was reproducible. This saves time, but wouldn't help the total issue count. At the very least, it could be introduced with the tags "(S1 - Need confirmation)" and "(S1 - Reproducible)". The stale bot could take 60 or 90 days if it has "(S1 - Reproducible)" as a tag. That seems like a nice middle ground between issues not being closed and getting attention.

My solution 1 would help with the "no one actively tried to confrm it" part. I've seen the bot that auto-requests reviews for PR's, and figured it might be of use here. If a review is requested after 15 days for a bug, in the simplest case, questions could be asked: "have you been able to reproduce this?", or a skim over the issue could reveal that the bug is reproducible and warranted a bug report, then get confirmed. In most cases, it could roll-over to just becoming stale after 30 days again with the stale-bot, but at least it got slight attention.

**Comment by Maleclypse** [comment]
You cannot confirm your own bugs.  You can ask in player help if anyone else can confirm a bug and then we can reopen it and mark it S2 if someone else can confirm it.

**Comment by Maleclypse** [comment]
So I just looked through your closed issues and you only have one stale closed bug report that might be valid.  So I feel like while you may have several that have recently entered stale there is a better than average chance they'll get confirmed before they get closed.

**Comment by harakka** [comment]
> If I put in the time to figure out how to reproduce a bug, give a full report, pictures, save file, version info, etc, it should at least get some amount of review. If it just gets closed as "not planned" after a little over 30 days, it seems like kind of a waste of time to have made it in the first place. Then what exactly are we supposed to do after that?

Unfortunately the nature of this being a volunteer project means that both bug reporting and issue triage are volunteer things with no-one guaranteed to do them nor having a guaranteed payoff. The problem is not lack of people being notified, but just lack of enough people with time and interest to go through all the bug reports the project gets, never mind all the other issues.

Issues closed as stale are still considered. It just helps us keep track of stuff that is currently being actively worked on. Users verifying their own issues is not going to be the solution.

**Comment by PatrikLundell** [comment]
The main problem is the lack of people who are both willing and able to fix the bugs reported. Most people who are able to and willing to work on the game are busy with their own stuff.

Do I have a solution to that problem? No.

The killer bot is good at keeping the number of open issues down both by killing them and by discouraging people to waste time on writing reports nobody will act on (and why waste time on confirming a report that won't be acted upon anyway). It obviously is useless as a means to deal with bugs, but that's not its purpose.

**Comment by kevingranade** [comment]
Marking bugs stale and closing them isn't the problem,  being unable to triage all the incoming issues is the problem.

If you have proposals for streamlining triage I'm all ears. 
If you want to join the triage team, reproduce bugs and mark them confirmed,  just ask and you'll most likely get those permissions if you've just been around a little while without making trouble.

As for the specific suggestions
Pinging contributors is not an option,  you can get feeds of issue creation or even sign up for a triage rotation https://www.codetriage.com/cleverraven/cataclysm-dda and a greater volume of pings isn't going to do anything to help. 
Tagging your own issue is helpful if it's simply informative, doing so to avoid stale bot doesn't work. 
A template that aids in reproduceability would be good,  but there's no way to validate that it actually works unless someone follows it and marks it confirmed.
It's not 30 days until close,  it's 60 days since the last interaction with the issue. That is an ETERNITY in dda terms, and the issue is already dead at that point,  the bot is just stating the obvious at that point. 


In short, you're not even trying to treat the symotom instead of the disease, you're trying to treat the side effects of the medicine for the disease but still not addressing either the symptoms or the disease.
